[Bug] Layerwise casting tests fail on MPS backend due to float8 unsupported dtype

### Describe the bug

When running the unit test suite on Apple Silicon (macOS) using the MPS backend (`torch_device = "mps"`), tests checking `enable_layerwise_casting` fail. This occurs because the MPS backend does not support the `torch.float8_e4m3fn` dtype.

Several test cases unconditionally configure layerwise casting with `storage_dtype=torch.float8_e4m3fn`, leading to a `TypeError: Trying to convert Float8_e4m3fn to the MPS backend but it does not have support for that dtype.` on MPS.

I intend to submit a pull request to fix this by dynamically skipping the tests when the active device doesn't support the `float8_e4m3fn` dtype.

### Reproduction

On an Apple Silicon Mac:
```bash
python3 -m pytest tests/pipelines/krea2/test_krea2.py -k test_layerwise_casting_inference
```

### Logs

```tb
=================================== FAILURES ===================================
___________ Krea2PipelineFastTests.test_layerwise_casting_inference ____________

self = <tests.pipelines.krea2.test_krea2.Krea2PipelineFastTests testMethod=test_layerwise_casting_inference>

    def test_layerwise_casting_inference(self):
        if not self.test_layerwise_casting:
            return
    
        components = self.get_dummy_components()
        pipe = self.pipeline_class(**components)
        pipe.to(torch_device, dtype=torch.bfloat16)
        pipe.set_progress_bar_config(disable=None)
    
        denoiser = pipe.transformer if hasattr(pipe, "transformer") else pipe.unet
>       denoiser.enable_layerwise_casting(storage_dtype=torch.float8_e4m3fn, compute_dtype=torch.bfloat16)

tests/pipelines/test_pipelines_common.py:2305: 
...
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/torch/nn/modules/module.py:1370: TypeError
E           TypeError: Trying to convert Float8_e4m3fn to the MPS backend but it does not have support for that dtype.
```

### System Info

- Diffusers version: 0.39.0.dev0
- Platform: macOS (Apple Silicon)
- Python version: 3.11.9
- PyTorch version: 2.11.0
- MPS available: True

### Who can help?

@DN6 @sayakpaul @yiyixuxu

--- a/tests/lora/utils.py
+++ b/tests/lora/utils.py
@@ -2105,12 +2105,12 @@ class PeftLoraLoaderMixinTests:
         self.assertTrue(not np.allclose(original_output, lora_output_diff_alpha, atol=1e-3, rtol=1e-3))
         self.assertTrue(not np.allclose(lora_output_diff_alpha, lora_output_same_rank, atol=1e-3, rtol=1e-3))
 
+    @pytest.mark.xfail(
+        condition=torch_device == "mps",
+        reason="MPS does not support float8 casting.",
+        strict=True,
+    )
     def test_layerwise_casting_inference_denoiser(self):
-        try:
-            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
-        except TypeError:
-            self.skipTest(f"Device {torch_device} does not support float8 storage dtype.")
-
         from diffusers.hooks._common import _GO_LC_SUPPORTED_PYTORCH_LAYERS
         from diffusers.hooks.layerwise_casting import DEFAULT_SKIP_MODULES_PATTERN
 
@@ -2154,6 +2154,11 @@ class PeftLoraLoaderMixinTests:
         pipe_float8_e4m3_bf16 = initialize_pipeline(storage_dtype=torch.float8_e4m3fn, compute_dtype=torch.bfloat16)
         pipe_float8_e4m3_bf16(**inputs, generator=torch.manual_seed(0))[0]
 
+    @pytest.mark.xfail(
+        condition=torch_device == "mps",
+        reason="MPS does not support float8 casting.",
+        strict=True,
+    )
     @require_peft_version_greater("0.14.0")
     def test_layerwise_casting_peft_input_autocast_denoiser(self):
         r"""
@@ -2169,10 +2174,6 @@ class PeftLoraLoaderMixinTests:
 
         See the docstring of [`hooks.layerwise_casting.PeftInputAutocastDisableHook`] for more details.
         """
-        try:
-            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
-        except TypeError:
-            self.skipTest(f"Device {torch_device} does not support float8 storage dtype.")
 
         from diffusers.hooks._common import _GO_LC_SUPPORTED_PYTORCH_LAYERS
         from diffusers.hooks.layerwise_casting import (

--- a/tests/lora/utils.py
+++ b/tests/lora/utils.py
@@ -2106,6 +2106,11 @@ class PeftLoraLoaderMixinTests:
         self.assertTrue(not np.allclose(lora_output_diff_alpha, lora_output_same_rank, atol=1e-3, rtol=1e-3))
 
     def test_layerwise_casting_inference_denoiser(self):
+        try:
+            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
+        except TypeError:
+            self.skipTest(f"Device {torch_device} does not support float8 storage dtype.")
+
         from diffusers.hooks._common import _GO_LC_SUPPORTED_PYTORCH_LAYERS
         from diffusers.hooks.layerwise_casting import DEFAULT_SKIP_MODULES_PATTERN
 
@@ -2164,6 +2169,10 @@ class PeftLoraLoaderMixinTests:
 
         See the docstring of [`hooks.layerwise_casting.PeftInputAutocastDisableHook`] for more details.
         """
+        try:
+            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
+        except TypeError:
+            self.skipTest(f"Device {torch_device} does not support float8 storage dtype.")
 
         from diffusers.hooks._common import _GO_LC_SUPPORTED_PYTORCH_LAYERS
         from diffusers.hooks.layerwise_casting import (

--- a/tests/models/testing_utils/memory.py
+++ b/tests/models/testing_utils/memory.py
@@ -383,13 +383,13 @@ class LayerwiseCastingTesterMixin:
         - get_dummy_inputs(): Returns dict of inputs to pass to the model forward pass
     """
 
+    @pytest.mark.xfail(
+        condition=torch_device == "mps",
+        reason="MPS does not support float8 casting.",
+        strict=True,
+    )
     @torch.no_grad()
     def test_layerwise_casting_memory(self):
-        try:
-            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
-        except TypeError:
-            pytest.skip(f"Device {torch_device} does not support float8 storage dtype.")
-
         MB_TOLERANCE = 0.2
         LEAST_COMPUTE_CAPABILITY = 8.0
 
@@ -441,12 +441,12 @@ class LayerwiseCastingTesterMixin:
             or abs(fp8_e4m3_fp32_max_memory - fp32_max_memory) < MB_TOLERANCE
         ), "Peak memory should be lower or within tolerance with fp8 storage"
 
+    @pytest.mark.xfail(
+        condition=torch_device == "mps",
+        reason="MPS does not support float8 casting.",
+        strict=True,
+    )
     def test_layerwise_casting_training(self):
-        try:
-            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
-        except TypeError:
-            pytest.skip(f"Device {torch_device} does not support float8 storage dtype.")
-
         def test_fn(storage_dtype, compute_dtype):
             if torch.device(torch_device).type == "cpu" and compute_dtype == torch.bfloat16:
                 pytest.skip("Skipping test because CPU doesn't go well with bfloat16.")

--- a/tests/models/testing_utils/memory.py
+++ b/tests/models/testing_utils/memory.py
@@ -385,6 +385,11 @@ class LayerwiseCastingTesterMixin:
 
     @torch.no_grad()
     def test_layerwise_casting_memory(self):
+        try:
+            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
+        except TypeError:
+            pytest.skip(f"Device {torch_device} does not support float8 storage dtype.")
+
         MB_TOLERANCE = 0.2
         LEAST_COMPUTE_CAPABILITY = 8.0
 
@@ -437,6 +442,11 @@ class LayerwiseCastingTesterMixin:
         ), "Peak memory should be lower or within tolerance with fp8 storage"
 
     def test_layerwise_casting_training(self):
+        try:
+            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
+        except TypeError:
+            pytest.skip(f"Device {torch_device} does not support float8 storage dtype.")
+
         def test_fn(storage_dtype, compute_dtype):
             if torch.device(torch_device).type == "cpu" and compute_dtype == torch.bfloat16:
                 pytest.skip("Skipping test because CPU doesn't go well with bfloat16.")

--- a/tests/pipelines/test_pipelines_common.py
+++ b/tests/pipelines/test_pipelines_common.py
@@ -2296,6 +2296,11 @@ class PipelineTesterMixin:
         if not self.test_layerwise_casting:
             return
 
+        try:
+            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
+        except TypeError:
+            self.skipTest(f"Device {torch_device} does not support float8 storage dtype.")
+
         components = self.get_dummy_components()
         pipe = self.pipeline_class(**components)
         pipe.to(torch_device, dtype=torch.bfloat16)

--- a/tests/pipelines/test_pipelines_common.py
+++ b/tests/pipelines/test_pipelines_common.py
@@ -2292,14 +2292,14 @@ class PipelineTesterMixin:
         elif isinstance(pipeline_out, torch.Tensor) and isinstance(loaded_pipeline_out, torch.Tensor):
             assert torch.allclose(pipeline_out, loaded_pipeline_out, atol=atol, rtol=rtol)
 
+    @pytest.mark.xfail(
+        condition=torch_device == "mps",
+        reason="MPS does not support float8 casting.",
+        strict=True,
+    )
     def test_layerwise_casting_inference(self):
         if not self.test_layerwise_casting:
             return
-
-        try:
-            torch.zeros(1, device=torch_device).to(dtype=torch.float8_e4m3fn)
-        except TypeError:
-            self.skipTest(f"Device {torch_device} does not support float8 storage dtype.")
 
         components = self.get_dummy_components()
         pipe = self.pipeline_class(**components)

